Vates/VatesAPI/src/vtkMDHexFactory.cpp

```diff
@@ -77,30 +76,30 @@ void vtkMDHexFactory::doCreate(
               << " boxes down to depth " << m_maxDepth << std::endl;
 
   // Create 8 points per box.
-  vtkPoints *points = vtkPoints::New();
+  vtkNew<vtkPoints> points;
   points->Allocate(numBoxes * 8);
   points->SetNumberOfPoints(numBoxes * 8);
 
   // One scalar per box
-  vtkFloatArray *signals = vtkFloatArray::New();
+  vtkNew<vtkFloatArray> signals;
   signals->Allocate(numBoxes);
   signals->SetName(ScalarName.c_str());
   signals->SetNumberOfComponents(1);
-  // signals->SetNumberOfValues(numBoxes);
 
   // To cache the signal
-  float *signalArray = new float[numBoxes];
+  std::vector<float> signalCache;
+  signalCache.reserve(numBoxes);
 
   // True for boxes that we will use
-  bool *useBox = new bool[numBoxes];
-  memset(useBox, 0, sizeof(bool) * numBoxes);
+  std::unique_ptr<bool[]> useBox(new bool[numBoxes]);
```

Comment:
I know this PR is still in progress, but I wanted to make sure you are aware of [Mantid::Kernel::make_unique](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/inc/MantidKernel/make_unique.h)